### PR TITLE
Sgt back end

### PR DESCRIPTION
**Pull request by CristoferMar**
sgt-back-end

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  if (!(req.body.name && req.body.course && req.body.score)) {
+    res.status(400).json({ error: 'name, course, & score are required fields' });
+    return;
+  }
+  const gradeId = parseInt(req.params.gradeId);
+  const score = parseInt(req.body.score);
+  if (!((score.toString().length === req.body.score.length && score > 0) &&
```
Comment (thebearingedge) [inline sgt-back-end/index.js:104]: i don't really understand why these `length` comparisons are necessary. does your code work as intended if you remove all of these?
```js
score.toString().length === req.body.score.length
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok, overall excellent, but there is a comparison throughout your code that i can't seem to see the need for

File: sgt-back-end/index.js (hunk 1 above)
Comment (CristoferMar) [reply sgt-back-end/index.js:104]: These lines make sure the numbers being passed are only integers.
Without the .lengths, someone can pass "api/grades/5LOL"
The ID would be 5.

To make sure this gets avoided, I'm comparing the length of the return of parseInt() to the length of the initial value.

But I can remove these lines if you believe they are unnecessary

Comment (thebearingedge) [reply sgt-back-end/index.js:104]: @CristoferMar There is a `Number.isInteger()` method to check that.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**